tonilopezmr/CaptchaPartTwo.kt

```diff
@@ -0,0 +1,58 @@
+package day1
+
+import org.junit.Assert.*
+import org.junit.Test
+
+
+class CaptchaPartTwo {
+
+    fun sequence(sequence: String) : Int {
+        if (sequence.length % 2 != 0) return 0
+
+        val displacement = sequence.length / 2
+        val captcha = sequence + sequence.substring(0, displacement)
+
+        return captcha.foldIndexed(0) { idx, acc, a ->
+            if (idx < sequence.length && a.toString().toInt() == captcha[idx + displacement].toString().toInt()) {
```

Comment:
What about captcha[(idx  + displacement)%size] ? That way you don't need line 13, although you do 1 more operation each iteraation 🤔

Reply:
Yes! That's right!